onadata/apps/api/viewsets/tableau_viewset.py

```diff
@@ -0,0 +1,304 @@
+        parent_table: str = None,
+        parent_id: int = None,
+        current_table: str = DEFAULT_TABLE_NAME):
+    result = []
+    if data:
+        for idx, row in enumerate(data, start=1):
+            flat_dict = defaultdict(list)
+            row_id = row.get('_id')
+
+            if not row_id and parent_id:
+                row_id = int(pairing(parent_id, idx))
+                flat_dict[PARENT_ID] = parent_id
+                flat_dict[PARENT_TABLE] = parent_table
+                flat_dict[ID] = row_id
+            else:
+                flat_dict[ID] = row_id
+
+            for (key, value) in row.items():
+                qstn = xform.get_element(key)
+                if qstn:
+                    qstn_type = qstn.get('type')
+                    qstn_name = qstn.get('name')
+
+                    prefix_parts = [
+                        question['name'] for question in qstn.get_lineage()
+                        if question['type'] == 'group'
+                    ]
+                    prefix = "_".join(prefix_parts)
+
+                    if qstn_type == REPEAT_SELECT_TYPE:
+                        repeat_data = process_tableau_data(
+                            value, xform,
+                            parent_table=current_table,
+                            parent_id=row_id,
+                            current_table=qstn_name)
+                        cleaned_data = unpack_repeat_data(
+                            repeat_data, flat_dict)
+                        flat_dict[qstn_name] = cleaned_data
+                    elif qstn_type == MULTIPLE_SELECT_TYPE:
+                        picked_choices = value.split(" ")
+                        choice_names = [
+                            question["name"] for question in qstn["children"]]
+                        list_name = qstn.get('list_name')
+                        select_multiple_data = unpack_select_multiple_data(
+                            picked_choices, list_name, choice_names, prefix)
+                        flat_dict.update(select_multiple_data)
+                    elif qstn_type == 'geopoint':
+                        gps_parts = unpack_gps_data(
+                            value, qstn_name, prefix)
+                        flat_dict.update(gps_parts)
+                    else:
+                        if prefix:
+                            qstn_name = f"{prefix}_{qstn_name}"
+                        flat_dict[qstn_name] = value
+            result.append(dict(flat_dict))
+    return result
+
+
+def unpack_select_multiple_data(picked_choices, list_name,
+                                choice_names, prefix):
+    unpacked_data = {}
+    for choice in choice_names:
+        qstn_name = f"{list_name}_{choice}"
+
+        if prefix:
+            qstn_name = prefix + '_' + qstn_name
+
+        if choice in picked_choices:
+            unpacked_data[qstn_name] = "TRUE"
+        else:
+            unpacked_data[qstn_name] = "FALSE"
+        return unpacked_data
```

Comment:
Shouldn't this return statement be outside the `for` loop?

Reply:
Guessing we have to loop through the entire `choice_names` list? Kindly add/update a test for this case too..._